lib/jekyll-relative-links/generator.rb

```diff
@@ -6,7 +6,7 @@ class Generator < Jekyll::Generator
     include Jekyll::Filters::URLFilters
 
     LINK_TEXT_REGEX = %r!([^\]]+)!
-    FRAGMENT_REGEX = %r!(#.+)?!
+    FRAGMENT_REGEX = %r!(#(?:[^)]+|.+$))?!
```

Comment:
Actually, I’ve just simplified the change, using `.+?` instead of `.+`. Much simpler to understand, the tests pass all the same.

Could you please take a second look?